File: backend/app/schemas/task.py
"""
任务相关数据模型
"""
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, Field


class TaskBase(BaseModel):
    """任务基础模型"""

    name: str = Field(..., description="任务名称")
    task_type: str = Field(..., description="任务类型: search_job, auto_apply, auto_chat等")
    config: dict = Field(default_factory=dict, description="任务配置")


class TaskCreate(TaskBase):
    """创建任务"""
    pass


class TaskUpdate(BaseModel):
    """更新任务"""
    status: Optional[str] = Field(None, description="状态")
    config: Optional[dict] = Field(None, description="任务配置")
    result: Optional[dict] = Field(None, description="任务结果")


class TaskResponse(TaskBase):
    """任务响应"""
    id: str = Field(..., description="任务ID")
    status: str = Field(default="pending", description="状态: pending, running, completed, failed")
    result: Optional[dict] = Field(None, description="任务结果")
    error_message: Optional[str] = Field(None, description="错误信息")
    created_at: datetime = Field(default_factory=datetime.now, description="创建时间")
    updated_at: datetime = Field(default_factory=datetime.now, description="更新时间")

    class Config:
        json_schema_extra = {
            "example": {
                "id": "123",
                "name": "自动投递任务",
                "task_type": "auto_apply",
                "status": "running",
                "config": {"keyword": "Python", "city": "北京"}
            }
        }

File: frontend/src/api/task.ts
/**
 * 任务相关API
 */
import { http, ApiResponse, PageResponse } from '@/utils/request'

// 任务接口
export interface Task {
  id: string
  name: string
  task_type: string
  config: Record<string, any>
  status: 'pending' | 'running' | 'completed' | 'failed'
  result?: Record<string, any>
  error_message?: string
  created_at: string
  updated_at: string
}

// 任务查询参数
export interface TaskQuery {
  page?: number
  page_size?: number
  status?: string
  task_type?: string
}

// 创建任务参数
export interface TaskCreate {
  name: string
  task_type: string
  config: Record<string, any>
}

// 更新任务参数
export interface TaskUpdate {
  status?: string
  config?: Record<string, any>
  result?: Record<string, any>
}

export const taskApi = {
  // 获取任务列表
  getList(params: TaskQuery) {
    return http.get<PageResponse<Task>>('/tasks', { params })
  },

  // 获取任务详情
  getDetail(id: string) {
    return http.get<Task>(`/tasks/${id}`)
  },

  // 创建任务
  create(data: TaskCreate) {
    return http.post<Task>('/tasks', data)
  },

  // 更新任务
  update(id: string, data: TaskUpdate) {
    return http.put<Task>(`/tasks/${id}`, data)
  },

  // 删除任务
  delete(id: string) {
    return http.delete(`/tasks/${id}`)
  },

  // 执行任务
  execute(id: string) {
    return http.post(`/tasks/${id}/execute`)
  },
}
